### ISSUE TITLE
byte-sized OnHeapStore.compute(3args) leaks if the 3rd arg returns FALSE and the expiration policy throws

### ISSUE THREAD

**Issue by lorban** [ehcache/ehcache3#823]
Assume that `OnHeapStore.compute(3args)` is called on an existing, not expired mapping. Also assume that the 2nd arg does not return null and the 3rd one returns `FALSE`.

This branch gets executed:
```
          } else if ((eq(existingValue, computedValue)) && (!replaceEqual.apply())) {
            if (mappedValue != null) {
              return setAccessTimeAndExpiryThenReturnMapping(key, mappedValue, now, eventSink);
            }
```

leaving `AtomicBoolean write` set to its default `false`. `setAccessTimeAndExpiryThenReturnMapping()` calls the expiration policy, and if the latter throws a `RuntimeException`, `null` is returned - which ends up being the computed method's return value which makes the backend remove the mapping.

`decrementCurrentUsageInBytesIfRequired` is only called when  `AtomicBoolean write` is set to true which won't happen here despite the fact that the mapping was dropped.

**Comment by ljacomet** [comment]
Most likely be resolved through #825

**Comment by ljacomet** [comment]
`computeIfPresent` suffered from the same issue
